shell/browser/printing/print_preview_message_handler.cc

```diff
@@ -107,14 +112,45 @@ void PrintPreviewMessageHandler::OnMetafileReadyForPrinting(
   }
 }
 
+void PrintPreviewMessageHandler::OnPrepareForDocumentToPdfDone(
+    const PrintHostMsg_PreviewIds& ids,
+    printing::mojom::PrintCompositor::Status status) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  if (status != printing::mojom::PrintCompositor::Status::kSuccess) {
+    LOG(ERROR) << "Preparing document for pdf failed with error " << status;
```

Comment:
Will `OnMetafileReadyForPrinting` be called even when this fails ? want to make sure we don't have an unresolved promise.

Reply:
yes - this is always triggered from local testing. if we try and reject before this, in page composition failure, it results in duplicate rejections